tests/cpgrid/LgrChecks.hpp

```diff
@@ -100,66 +109,105 @@ void checkExpectedVertexGlobalIdsCount(const Dune::CpGrid& grid,
 
+void checkEqualVertexFaceCellSizes(const Dune::CpGrid& grid,
+                                   const Dune::CpGrid& other_grid);
 
-} // namespace Opm
+void checkEqualVertexGeometrySet(const Dune::CpGrid& grid,
+                                 const Dune::CpGrid& other_grid);
+
+void checkEqualCellGeometrySet(const Dune::CpGrid& grid,
+                               const Dune::CpGrid& other_grid);
+
+template<typename GridView>
+void checkEqualIntersectionsGeometry(const GridView& grid_view,
+                                     const Dune::cpgrid::Entity<0>& element,
+                                     const GridView& equiv_grid_view,
+                                     const Dune::cpgrid::Entity<0>& other_grid_elem);
+
+void checkLeafGridGeometryEquality(const Dune::CpGrid& grid,
+                                   const Dune::CpGrid& other_grid);
 
+template<typename T>
+bool areClose(const T& cont1,
+              const T& cont2);
 
+void adaptGridWithParams(Dune::CpGrid& grid,
+                         const std::array<int,3>& cells_per_dim,
+                         const std::vector<int>& markedCells);
 
+void adaptGrid(Dune::CpGrid& grid,
+               const std::vector<int>& markedCells);
 
-void Opm::checkAverageCenterAndVolume(Dune::cpgrid::HierarchicIterator it,
-                                      const Dune::cpgrid::HierarchicIterator& endIt,
-                                      double total_children)
+} // namespace Opm
+
+void Opm::checkReferenceElemParentCellVolume(Dune::cpgrid::HierarchicIterator it,
+                                             const Dune::cpgrid::HierarchicIterator& endIt)
 {
-    double referenceElemOneParent_volume_it{};
-    std::array<double,3> referenceElem_entity_center_it; // Expected {.5,.5,.5}
+    double reference_elem_parent_cell_volume = std::accumulate(it, endIt, 0.0,
+                                                               [](double sum, const Dune::cpgrid::Entity<0>& child) {
+                                                                   return sum + child.geometryInFather().volume();
+                                                               });
+    BOOST_CHECK_CLOSE(reference_elem_parent_cell_volume, 1.0, 1e-13);
+}
 
-    for (; it != endIt; ++it)
-    {
-        referenceElemOneParent_volume_it += it-> geometryInFather().volume();
-        for (int c = 0; c < 3; ++c)
-        {
-            referenceElem_entity_center_it[c] += (it-> geometryInFather().center())[c];
+void Opm::checkReferenceElemParentCellCenter(Dune::cpgrid::HierarchicIterator it,
+                                             const Dune::cpgrid::HierarchicIterator& endIt,
+                                             double total_children)
+{
+    assert(total_children);
+    std::array<double,3> reference_elem_parent_cell_center = {0.0, 0.0, 0.0}; // Expected {.5,.5,.5}
+    for (; it != endIt; ++it) {
+        for (int c = 0; c < 3; ++c) {
```

Comment:
Maybe should add an assert that all elements are on the same level as this is assumed by the algorithm?